presto-hive/src/main/java/io/prestosql/plugin/hive/s3/S3SecurityMappingConfig.java

```diff
@@ -27,6 +27,20 @@
     private String roleCredentialName;
     private Duration refreshPeriod;
     private String colonReplacement;
+    private String securityMappingsProvider;
+
+    public String getSecurityMappingsProvider()
+    {
+        return securityMappingsProvider == null ? FileBasedS3SecurityMappingsProvider.class.getName() : securityMappingsProvider;
+    }
+
+    @Config("hive.s3.security-mapping.provider")
+    @ConfigDescription("class used to obtain the s3 security mappings")
```

Comment:
We do not use reflection for things like that. We need to have all provider implementations here.

Reply:
So what I did locally to test the implementation and to setup for later is: I set up a new maven project, added `presto-hive` as a dependency and implemented a custom `S3SecurityMappingsProvider` in that project. Then I built the jar and put it into the hive-hadoop2 plugin directory. That way it can be found on the classpath and the custom implementation can be used. I think it leaves you more flexible in terms of implementation.